### PR TITLE
Release scripts fixes

### PR DESCRIPTION
**Pull request by mkuznyetsov**
### What does this PR do?
Apply several fixes to the problems that were discovered during the last release

### CODE REVIEW

File: .github/workflows/release.yml [hunk 1]
```diff
@@ -31,12 +31,12 @@ jobs:
...
+          curl -s "https://raw.githubusercontent.com/kubernetes-sigs/kustomize/kustomize/v4.0.4/hack/install_kustomize.sh"  | bash
           cd ${GITHUB_WORKSPACE}
       - name: Build image, generate deployments and push git tag
         run: |
           git config --global user.name "Mykhailo Kuznietsov"
           git config --global user.email "mkuznets@redhat.com"
 
           export GITHUB_TOKEN=${{ secrets.CHE_INCUBATOR_BOT_GITHUB_TOKEN }}
-          ./make-release.sh --version ${{ github.event.inputs.version}}
+          bash -x ./make-release.sh --version ${{ github.event.inputs.version}}
```
Comment (sleshchenko) [inline .github/workflows/release.yml:42]: is -x here just for debugging or better to always has it?

File: make-release.sh [hunk 2]
```diff
@@ -31,45 +31,44 @@ bump_version () {
...
+    hub pull-request -f -m "${lastCommitComment}" -b "${BUMP_BRANCH}" -h "${PR_BRANCH}"	
+  fi 	
+  set -e	
+  git checkout "${CURRENT_BRANCH}"	
+}	
+
+usage ()	
+{	
+  echo "Usage: $0 --version [VERSION TO RELEASE]"	
+  echo "Example: $0 --version 0.1.0"; echo
```
Comment (sleshchenko) [inline make-release.sh:60]:
```suggestion
  echo "Example: $0 --version v0.1.0"; echo	
```
?

File: .github/workflows/release.yml (hunk 1 above)
Comment (mkuznyetsov) [reply .github/workflows/release.yml:42]: tbf I can remove it from here for now

File: make-release.sh [hunk 3]
```diff
@@ -31,45 +31,44 @@ bump_version () {
...
+    hub pull-request -f -m "${lastCommitComment}" -b "${BUMP_BRANCH}" -h "${PR_BRANCH}"	
+  fi 	
+  set -e	
+  git checkout "${CURRENT_BRANCH}"	
+}	
+
+usage ()	
+{	
+  echo "Usage: $0 --version [VERSION TO RELEASE]"	
+  echo "Example: $0 --version v0.1.0"; echo
```
Comment (sleshchenko) [inline make-release.sh:60]:
```suggestion
  echo "Example: $0 --version v0.1.0\n";
```

File: make-release.sh [hunk 4]
```diff
@@ -31,45 +31,44 @@ bump_version () {
...
+  if [[ $? -gt 0 ]] || [[ $PUSH_TRY == *"protected branch hook declined"* ]]; then	
+    PR_BRANCH=pr-${BUMP_BRANCH}-to-${NEXT_VERSION}	
+    # create pull request for the main branch branch, as branch is restricted	
+    git branch "${PR_BRANCH}"	
+    git checkout "${PR_BRANCH}"	
+    git pull origin "${PR_BRANCH}"	
+    git push origin "${PR_BRANCH}"	
+    lastCommitComment="$(git log -1 --pretty=%B)"	
+    hub pull-request -f -m "${lastCommitComment}" -b "${BUMP_BRANCH}" -h "${PR_BRANCH}"	
+  fi
```
Comment (sleshchenko) [inline make-release.sh:52]: I don't know which editor you use, but it's terrible it provides `tab` symbol in the end of each line. Please remove

File: make-release.sh [hunk 5]
```diff
@@ -33,43 +33,41 @@ bump_version () {
...
+  # shellcheck disable=SC2181
+  if [[ $? -gt 0 ]] || [[ $PUSH_TRY == *"protected branch hook declined"* ]]; then
+    PR_BRANCH=pr-${BUMP_BRANCH}-to-${NEXT_VERSION}
+    # create pull request for the main branch branch, as branch is restricted
+    git branch "${PR_BRANCH}"
+    git checkout "${PR_BRANCH}"
+    git pull origin "${PR_BRANCH}"
+    git push origin "${PR_BRANCH}"
+    lastCommitComment="$(git log -1 --pretty=%B)"
+    hub pull-request -f -m "${lastCommitComment}" -b "${BUMP_BRANCH}" -h "${PR_BRANCH}"
```
Comment (sleshchenko) [inline make-release.sh:51]: That seems to be a good time to switch to `gh` CLI but it's not mandatory for this PR.

**Review by sleshchenko — COMMENTED** [review]
LGTM

**Review by sleshchenko — APPROVED** [review]

**Review by JPinkney — APPROVED** [review]

**Pull request merged**